iml-agent/src/action_plugins/stratagem/action_cloudsync.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+extern crate futures;
+extern crate tokio;
+
+use crate::agent_error::{ImlAgentError, RequiredError};
+use crate::lustre::search_rootpath;
+use iml_wire_types::{FidError, FidItem};
+use liblustreapi::LlapiFid;
+use tokio::process::Command;
+use tracing::error;
+
+fn get_unique(input: &str) -> String {
+    let mut output = String::new();
+    for c in input.chars() {
+        match c {
+            '/' => output.push_str("_"),
+            _ => output.push(c),
+        }
+    }
+    output
+}
+
+async fn archive_fids(
+    llapi: LlapiFid,
+    target_name: &String,
+    fid_list: Vec<FidItem>,
+) -> Result<Vec<FidError>, ImlAgentError> {
+    let mut result = Vec::new();
+
+    for fid in fid_list {
+        let fid_path = llapi.fid2path(&fid.fid)?;
+        let src_file = format!("{}/{}\n", llapi.mntpt(), &fid_path);
+
+        let output = Command::new("stgm_cloudsync")
+            .arg("push")
+            .arg(src_file)
+            .arg(format!("{}:{}", target_name, get_unique(&fid_path)))
+            .output();
+        let output = output.await?;
+
+        let res = FidError {
+            fid: fid.fid.clone(),
+            data: fid.data.clone(),
+            errno: output.status.code().unwrap_or(0) as i16,
+        };
+        result.push(res);
+
+        error!(
+            "exited with {} {} {}",
+            output.status,
+            std::str::from_utf8(&output.stdout)?,
+            std::str::from_utf8(&output.stderr)?
+        );
+    }
+
+    Ok(result)
+}
+
+async fn restore_fids(
+    llapi: LlapiFid,
+    target_name: &String,
+    fid_list: Vec<FidItem>,
+) -> Result<Vec<FidError>, ImlAgentError> {
+    let mut result = Vec::new();
+
+    for fid in fid_list {
+        let fid_path = llapi.fid2path(&fid.fid)?;
+        let src_file = format!("{}/{}\n", llapi.mntpt(), &fid_path);
+
+        let output = Command::new("stgm_cloudsync")
+            .arg("pull")
+            .arg(format!("{}:{}", target_name, get_unique(&fid_path)))
+            .arg(src_file)
+            .output();
+        let output = output.await?;
+
+        let res = FidError {
+            fid: fid.fid.clone(),
+            data: fid.data.clone(),
+            errno: output.status.code().unwrap_or(0) as i16,
+        };
+        result.push(res);
```

Comment:
```suggestion
        let FidItem { fid, data } = fid;

        let res = FidError {
            fid,
            data,
            errno: output.status.code().unwrap_or(0) as i16,
        };

        result.push(res);
```

You can use destructuring to move the values out, which will avoid the clone.

Reply:
See previous comment

Reply:
This will move the value, and will not clone.

Reply:
Done